lib/potluck.rb

```diff
@@ -0,0 +1,78 @@
+require_relative './dish'
+
+class Potluck
+
+  attr_reader   :date,
+                :dishes
+
+  def initialize(date)
+    @date = date
+    @dishes = []
+  end
+
+  def add_dish(dish)
+    @dishes << dish
+  end
+
+  def get_all_from_category(category_name)
+    dishes.find_all do |dish|
+      dish.category == category_name
+    end
+  end
+
+  def menu_hash(appetizer_names, entre_names, dessert_names)
+    {:appetizers => appetizer_names,
+      :entres => entre_names,
+      :desserts => dessert_names
+    }
+  end
+
+  def menu
+    dishes = @dishes
+    appetizer_names = apps(dishes)
+    entre_names = entres(dishes)
+    dessert_names = desserts(dishes)
+    menu_hash(appetizer_names, entre_names, dessert_names)
+  end
+
+  def apps(dishes)
```

Comment:
Instead of writing these 3 methods to get all the appetizers, desserts, and entrees, you can call your get_all_from_category method